lib/buffer.js

```diff
@@ -326,13 +326,34 @@ Object.defineProperty(Buffer.prototype, 'offset', {
 function slowToString(encoding, start, end) {
   var loweredCase = false;
 
-  start = start >>> 0;
-  end = end === undefined || end === Infinity ? this.length : end >>> 0;
+  if (start >= this.length || end <= 0 || Number.isNaN(end))
```

Comment:
On the other pull request you indicated that you wanted the same behavior as typed arrays

```javascript
let ab = new ArrayBuffer(8);
new Uint8Array(ab, 0, false).length === 0;
new Uint8Array(ab, 0, null).length === 0;
new Uint8Array(ab, 0, NaN).length === 0;
new Uint8Array(ab, 0, undefined).length === 8;
```

Assigning the default is normally captured by the check below `end > this.length || end >>> 0 !== parseInt(+end)` but `end >>> 0 !== parseInt(+end)` is `false` for `end = null` and `end = false` and is `true` for `end = undefined` and `end = NaN`. To ensure that the default value is not used when `end = NaN`, I special cased it above. If there is a more elegant check that could be used below that would be nicer.